lib/vocabularies/code.ts

```diff
@@ -90,8 +92,23 @@ export function callValidateCode(
   return context !== nil ? _`${func}.call(${context}, ${args})` : _`${func}(${args})`
 }
 
-export function usePattern({gen, it: {opts}}: KeywordCxt, pattern: string): Name {
+export function usePattern({gen, it: {self, opts}}: KeywordCxt, pattern: string): Name {
   const u = opts.unicodeRegExp ? "u" : ""
+  const useRe2 = opts.useRe2
+  if (u === "u" && useRe2) {
```

Comment:
Can you comment on the reason to restrict the RE2 protection to only the unicodeRegExp?

Reply:
RE2 always parses regexes as Unicode. The unicodeRegExp option is true by default, which means a developer must knowingly change that option. We could use RE2 nevertheless, but in those cases, RE2 will most likely not behave as the developer anticipates. 

In that case, we can convert the regex to be non-Unicode, but I do not know how to implement this as of now.